org.mwc.debrief.pepys/src/main/java/org/mwc/debrief/pepys/model/bean/FilterableBean.java

```diff
@@ -0,0 +1,24 @@
+/*******************************************************************************
+ * Debrief - the Open Source Maritime Analysis Application
+ * http://debrief.info
+ *
+ * (C) 2000-2020, Deep Blue C Technology Ltd
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the Eclipse Public License v1.0
+ * (http://www.eclipse.org/legal/epl-v10.html)
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *******************************************************************************/
+
+package org.mwc.debrief.pepys.model.bean;
+
```

Comment:
@saulhidalgoaular - for us to do this filter in Java, we have to load all the rows, and then apply the filter - on the client.

I _hope_ we can apply the filter at the server, and only get the datafiles that contain data matching the field.  Then, when the user selects "Import data" we get all the rows, but SQL filtered (_where_) according to time period or area of coverage.  Then, we only pull data onto the client (deserialise) that is actually going to be loaded into Debrief.

Happy to talk this through further, if necessary.

Reply:
Correct. My idea was to filter on the server. I created this interface to have access to the field names to prepare the query for the server.

Reply:
Aah, that's clever :-D